### PR TITLE
Add _foreach_addc(div/mul)(_).Tensor

### PR DESCRIPTION
**Pull request by cpuhrsch**
Support passing value scalars as a flat 1D Tensor.

Currently we can only pass either an individual scalar or a ScalarList.

### CODE REVIEW

File: test/test_foreach.py [hunk 1]
```diff
@@ -302,9 +306,12 @@ def test_pointwise_op_fastpath(self, device, dtype, op):
             self._test_pointwise_op(device, dtype, op, N, True, disable_fastpath)
             for scalar in Scalars:
                 self._test_pointwise_op(device, dtype, op, N, True, disable_fastpath, values=scalar)
-            for _, scalarlist in getScalarLists(N):
+                self._test_pointwise_op(device, dtype, op, N, True, disable_fastpath, values=torch.tensor(scalar))
```
Comment (albanD) [inline test/test_foreach.py:309]: Should you also test the case where `torch.tensor(scalar, device=device)` ?
Same for all the updates below.

File: aten/src/ATen/native/native_functions.yaml [hunk 2]
```diff
@@ -9838,6 +9838,14 @@
     CUDA: foreach_tensor_addcdiv_scalarlist_cuda_
   autogen: _foreach_addcdiv.ScalarList_out
 
+- func: _foreach_addcdiv_.Tensor(Tensor(a!)[] self, Tensor[] tensor1, Tensor[] tensor2, Tensor scalars) -> ()
```
Comment (albanD) [inline aten/src/ATen/native/native_functions.yaml:9841]: Curious why this is not `Tensor[] scalars`? That sounds more consistent?

File: test/test_foreach.py (hunk 1 above)
Comment (cpuhrsch) [reply test/test_foreach.py:309]: Yes, we'll need to add more "assertRegexFail" type tests

File: aten/src/ATen/native/native_functions.yaml (hunk 2 above)
Comment (cpuhrsch) [reply aten/src/ATen/native/native_functions.yaml:9841]: Because those are more difficult to convert to scalars efficiently.

Comment (cpuhrsch) [reply aten/src/ATen/native/native_functions.yaml:9841]: Well it can convert the step sizes to a torch.Tensor before calling into foreach

Comment (albanD) [reply aten/src/ATen/native/native_functions.yaml:9843]: Doing a .cat() before every call sounds like a big overhead?

Comment (ngimel) [reply aten/src/ATen/native/native_functions.yaml:9843]: It's for 1-element tensors on cpu only, should be fast.

**Review by ngimel — APPROVED** [review]

**Review by albanD — APPROVED** [review]
Sounds good!

**Pull request closed without merging**